tests/protocol/test_hostplugin.py

```diff
@@ -365,15 +399,159 @@ def test_validate_get_extension_artifacts(self):
             for k in expected_headers:
                 self.assertTrue(k in actual_headers)
                 self.assertEqual(expected_headers[k], actual_headers[k])
-    
+
+    @patch("azurelinuxagent.common.utils.restutil.http_get")
+    def test_health(self, patch_http_get):
```

Comment:
>test_health [](start = 8, length = 11)

I think there should be a test case for HTTP 502.  I think there should be a test case of /extensionArtifact and /vmAgentLog.  What else am I missing?

Reply:
There are quite a few tests still coming.